### DISCUSSION TITLE
Trigger dialog from sheet

### DISCUSSION THREAD

**Discussion by christoffer-nextase** [shadcn-ui/ui#1094] [Q&A] [answered]
How can i trigger a dialog and close the sheet component at the same time?

I have tried to set it up like this, but the sheet doesn't close:

Login.tsx
 ```typescript 
<Dialog>
   {trigger} 
    <DialogContent className="sm:max-w-[425px]">
      <DialogHeader>
        <DialogTitle>Logga in</DialogTitle>
      </DialogHeader>
      <form
        onSubmit={handleSubmit}
        className="mt-8 space-y-6"
        action="#"
        method="POST"
      >
        <LoginFormBody loading={loading} />
      </form>
    </DialogContent>
</Dialog>
```

LoginBar.tsx
 ```typescript 
<Login>
  <SheetClose asChild>
    <DialogTrigger className="cursor-pointer">
      <Button variant="default" size="sm" className="flex-1">
        Logga in
      </Button>
    </DialogTrigger>
  </SheetClose>
</Login>
```

Current result:
<img width="775" alt="Skärmavbild 2023-08-01 kl  14 54 55" src="https://github.com/shadcn-ui/ui/assets/126861351/3f12498e-e6f4-4576-be9c-6f4e1fd86fdb">

When the login button is clicked inside the sheet:
<img width="771" alt="Skärmavbild 2023-08-01 kl  14 55 15" src="https://github.com/shadcn-ui/ui/assets/126861351/821dc272-64fc-43cc-9c6a-9fb262ebde08">

**Comment by limitless-dev** [comment, marked as answer]
You can handle that manually with the `open` and `onOpenChange` props in the `Sheet` component.

```tsx
const [sheetOpen, setSheetOpen] = useState<boolean>(false);
```
```tsx
<Sheet open={sheetOpen} onOpenChange={setSheetOpen}>
{/*   */}
</Sheet>
```

Now when ever the Login is open, you do `setSheetOpen(false)`


Here is an example I used to close menu drop down when use clicks on edit (which opens a sheet):
```tsx
<Sheet
  open={sheetOpen}
  onOpenChange={() => {
    setSheetOpen(!sheetOpen);
    setMenuOpen?.(!sheetOpen);
    }}
>
{/*   */}
</Sheet>
```
